packages/kbn-alerts-ui-shared/src/rule_type_modal/components/rule_type_modal.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { useCallback } from 'react';
+import {
+  EuiPageHeader,
+  EuiModal,
+  EuiPanel,
+  EuiPageHeaderSection,
+  EuiTitle,
+  EuiFieldSearch,
+  EuiSpacer,
+  useEuiTheme,
+  useCurrentEuiBreakpoint,
+  EuiEmptyPrompt,
+  EuiLoadingSpinner,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import { RuleTypeList } from './rule_type_list';
+import { RuleTypeWithDescription, RuleTypeCountsByProducer } from '../types';
+
+export interface RuleTypeModalProps {
+  onClose: () => void;
+  onSelectRuleType: (ruleTypeId: string) => void;
+  onFilterByProducer: (producer: string | null) => void;
+  onChangeSearch: (search: string) => void;
+  searchString: string;
+  selectedProducer: string | null;
+}
+
+export interface RuleTypeModalState {
+  ruleTypes: RuleTypeWithDescription[];
+  ruleTypesLoading: boolean;
+  ruleTypeCountsByProducer: RuleTypeCountsByProducer;
+}
+
+export const RuleTypeModal: React.FC<RuleTypeModalProps & RuleTypeModalState> = ({
+  onClose,
+  onSelectRuleType,
+  onFilterByProducer,
+  onChangeSearch,
+  ruleTypes,
+  ruleTypesLoading,
+  ruleTypeCountsByProducer,
+  searchString,
+  selectedProducer,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const currentBreakpoint = useCurrentEuiBreakpoint() ?? 'm';
+  const isFullscreenPortrait = ['s', 'xs'].includes(currentBreakpoint);
+
+  const loadingPrompt = (
+    <EuiEmptyPrompt title={<h2>Loading rule types</h2>} icon={<EuiLoadingSpinner size="xl" />} />
+  );
```

Comment:
translation, also let's move this out of this component if it doesn't depend on any component state or props